index.js

```diff
@@ -33,36 +35,33 @@ var WinstonDynamoDB = function (options) {
     this.logEvents = [];
     this.errorHandler = options.errorHandler;
 
-    if (options.dynamoDB) {
-        this.dynamoDB = options.dynamoDB;
-    } else {
-        if (this.proxyServer) {
-            AWS.config.update({
-                httpOptions: {
-                    agent: require('proxy-agent')(this.proxyServer)
-                }
-            });
-        }
-
-        var config = {};
+    let config = {};
 
-        if (awsAccessKeyId && awsSecretKey && awsRegion) {
-            config = {credentials: {accessKeyId: awsAccessKeyId, secretAccessKey: awsSecretKey}, region: awsRegion};
-        } else if (awsRegion && !awsAccessKeyId && !awsSecretKey) {
-            // Amazon SDK will automatically pull access credentials
-            // from IAM Role when running on EC2 but region still
-            // needs to be configured
-            config = { region: awsRegion };
-        }
+    if (this.proxyServer) {
+        const agent = require('proxy-agent')(this.proxyServer);
+        config.requestHandler = new NodeHttpHandler({
+            httpAgent: agent,
+            httpsAgent: agent
```

Comment:
NodeHttpHandler with https?

Reply:
They distinguish now between http and https, so their best practice it so include both